hibernate-core/src/test/java/org/hibernate/graph/EntityGraphDynamicBatchStyleTest.java

```diff
@@ -0,0 +1,143 @@
+package org.hibernate.graph;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+import javax.persistence.ManyToOne;
+import javax.persistence.OneToMany;
+import javax.persistence.Table;
+
+import org.hibernate.Hibernate;
+import org.hibernate.cfg.AvailableSettings;
+import org.hibernate.jpa.test.BaseEntityManagerFunctionalTestCase;
+import org.hibernate.loader.BatchFetchStyle;
+
+import org.hibernate.testing.TestForIssue;
+import org.junit.Before;
+import org.junit.Test;
+
+import org.hamcrest.CoreMatchers;
+
+import static javax.persistence.CascadeType.MERGE;
+import static javax.persistence.CascadeType.PERSIST;
+import static javax.persistence.CascadeType.REMOVE;
+import static org.hamcrest.CoreMatchers.*;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hibernate.testing.transaction.TransactionUtil2.inTransaction;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * @author David Hoffer
+ * @author Nathan Xu
+ */
+@TestForIssue( jiraKey = "HHH-14312" )
+public class EntityGraphDynamicBatchStyleTest extends BaseEntityManagerFunctionalTestCase {
+	private static final int BATCH_SIZE = 5;
+	private static final int NUM_OF_LOCATIONS = (BATCH_SIZE * 2) + 1;
+
+	@Override
+	protected Class<?>[] getAnnotatedClasses() {
+		return new Class<?>[] {
+				Fruit.class,
+				FruitLocation.class
+		};
+	}
+
+	@Override
+	protected void addConfigOptions(Map options) {
+		options.put( AvailableSettings.BATCH_FETCH_STYLE, BatchFetchStyle.DYNAMIC );
+		options.put( AvailableSettings.DEFAULT_BATCH_FETCH_SIZE, BATCH_SIZE);
```

Comment:
Probably the missing space before the right parenthesis is the checkstyle issue.

Reply:
Local checkstyle report points to the common `package-info.java` issue that new line is missing at the end of the file.